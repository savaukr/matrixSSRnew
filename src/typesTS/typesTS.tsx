import {
  ADD_ROW,
  DELETE_ROW,
  INCREASE_AMOUNT,
  MOUSE_OVER_CEIL,
  MOUSE_OVER_SUM,
  MOUSE_OUT,
  ADD_PARAMS,
  ADD_MATRIX
} from "../redux/types";

export type TFunc = (params:IStateParamsHelp) => void

export interface ICeil {
  id: string;
  amount: number;
}
export interface ICeilsById {
  [key:string]:ICeil;
}
export interface ICeils{
  allIds: string[];
  byId:  ICeilsById;
}

export interface IRow {
  id: string;
  ceils: string[] ;
}
export interface IRowsById {
  [key:string]: IRow;
}
export interface IRows {
  allIds: string[];
  byId:  IRowsById;
} 

export interface IMatrix {
  bright: string[] | [];
  ceils: ICeils;
  rows: IRows;
}
export interface IMatrixRow {
  ceils: ICeils;
  rows: IRows;
}
export interface IStateMatrix {
  matrix: IMatrix;
  params: IStateParamsHelp;
}

export interface IAverage {
  id: string;
  amount: number;
}

//interface for params 
export interface IStateParamsHelp {
  M1:number | null;
  N1:number | null;
  X1:number | null;

}

//interface for action

interface IOverCeilPayload {
  ceilId: string;
  X: number;
}

interface IAddRow {
  type: typeof ADD_ROW;
  payload: any;
}
interface IDeleteRow {
  type: typeof DELETE_ROW;
  payload: number;
}
interface IIncreaseAmount {
  type: typeof INCREASE_AMOUNT;
  payload: string;
}
interface IMouseOverCeil {
  type: typeof MOUSE_OVER_CEIL;
  payload: IOverCeilPayload;
}
interface IMouseOut {
  type: typeof MOUSE_OUT;
  payload: string;
}
interface IMouseOverSum {
  type: typeof MOUSE_OVER_SUM;
  payload:number;
}

interface IAddParams {
  type: typeof ADD_PARAMS;
  payload:  IStateParamsHelp;
}
interface IAddMatrix {
  type: typeof ADD_MATRIX;
  payload: IMatrix
}

export type ActionsTypes =
  | IAddRow
  | IDeleteRow
  | IIncreaseAmount
  | IMouseOverCeil
  | IMouseOut
  | IMouseOverSum
  | IAddParams
  | IAddMatrix
