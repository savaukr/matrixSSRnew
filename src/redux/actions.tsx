import {ADD_ROW, DELETE_ROW, INCREASE_AMOUNT,
		MOUSE_OVER_CEIL, MOUSE_OUT, MOUSE_OVER_SUM, ADD_PARAMS, ADD_MATRIX} from './types';
import { ActionsTypes, IMatrix, IStateParamsHelp } from '../typesTS/typesTS'

export  function addRow(matrix: IMatrix): ActionsTypes {
	return {
		type: ADD_ROW,
		payload: matrix
	}
}

export function deleteRow(ind:number): ActionsTypes {
	return {
		type: DELETE_ROW,
		payload: ind
	}
}
export function increaseAmount(row:number, column:number):ActionsTypes {
	return {
		type: INCREASE_AMOUNT,
		payload: {row, column}
	}
}
export function mouseOverCeil(newState: IMatrix): ActionsTypes {
	return {
		type: MOUSE_OVER_CEIL,
		payload: newState
	}
}
export function mouseOut(newState: IMatrix): ActionsTypes {
	return {
		type: MOUSE_OUT,
		payload: newState
	}
}

export function mouseOverSum(newState: IMatrix): ActionsTypes {
	return {
		type: MOUSE_OVER_SUM,
		payload: newState
	}
}

export function addParams(params:IStateParamsHelp): ActionsTypes {
	return {
		type: ADD_PARAMS,
		payload: params
	}
}
export function addMatrix(newMatrix: IMatrix): ActionsTypes {
	return {
		type: ADD_MATRIX,
		payload: newMatrix
	}
}
