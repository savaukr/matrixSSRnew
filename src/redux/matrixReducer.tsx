import {ADD_ROW, DELETE_ROW, INCREASE_AMOUNT,
		MOUSE_OVER_CEIL, MOUSE_OUT, MOUSE_OVER_SUM, ADD_MATRIX} from './types'
import {M, N} from '../config/config'
import {  IMatrix, ActionsTypes } from '../typesTS/typesTS'
import {getMatrix, deleteRow} from '../matrixService/matrixService'



 const initialState:IMatrix = getMatrix(M, N)


export const matrixReducer = (state=initialState, action: ActionsTypes): IMatrix => {
	//let copyState = {...state}

	switch (action.type) {
		case ADD_ROW: 
			return { ...state, ...action.payload}
		case DELETE_ROW:
			return {...state, ...deleteRow(action.payload, {...state.rows}, {...state.ceils})}
		case INCREASE_AMOUNT: 
			const newCeils = {...state.ceils}
			newCeils.byId[action.payload].amount =  newCeils.byId[action.payload].amount+1
			return { ...state, ceils: newCeils}
		case MOUSE_OVER_CEIL: 
			return { ...state, ...action.payload}
		case MOUSE_OUT: 
			return { ...state, ...action.payload}
		case MOUSE_OVER_SUM: 
			return { ...state, ...action.payload}
		case ADD_MATRIX:
			return {...state, ...action.payload}

		default: return state
	}
}