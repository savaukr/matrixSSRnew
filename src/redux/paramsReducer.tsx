
import {  IStateParamsHelp, ActionsTypes } from '../typesTS/typesTS'
import {ADD_PARAMS} from './types'


 const initialState:IStateParamsHelp = {
	M1: null,
	N1:null,
	X1:null,
}

export const paramsReducer = (state=initialState, action: ActionsTypes): IStateParamsHelp => {
	
	 switch (action.type) {
		case ADD_PARAMS: 
			return { ...state, ... action.payload }
	 	default: return state
	}
	return state
}