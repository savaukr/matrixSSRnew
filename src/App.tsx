import React, { FC } from "react";
import { connect } from "react-redux";
import { AddRow } from "./components/AddRow/AddRow";
import { addRow, addParams, addMatrix } from "./redux/actions";
import Matrix from "./components/Matrix/Matrix";
import {FormParamsMatrix} from './components/FormParamsMatrix/FormParamsMatrix'
import { ActionsTypes, IStateParamsHelp, IMatrixRow, IMatrix, IStateMatrix } from "./typesTS/typesTS";
import { getMatrix } from "./matrixService/matrixService";


interface IAppProps {
  // addRow(row: IMatrixRow): ActionsTypes;
  // addParams(params: IStateParamsHelp): ActionsTypes;
  // addMatrix(newMatrix: IMatrix): ActionsTypes;
  matrix: IMatrix;
  params: IStateParamsHelp;
}

const App: FC<IAppProps> = ({ matrix, params }): any => {
  
  // function getMatrixRow(columns:number, i: number) {
  //   const row = [];
  //   for (let j = 0; j < columns; j++) {
  //     const amount = Math.floor(Math.random() * 1001);
  //     row[j] = { id: `${i}x${j}`, amount, bright: false, part: false };
  //   }
  //   return row;
  // }

  const addRowHandle = (event: React.MouseEvent<HTMLButtonElement>) => {
    // addRow(getMatrixRow();
    console.log('add row')
  };

  const addParamsHandle = ():any => {
    return (newParams: IStateParamsHelp) => {
      addParams(newParams)
      addMatrix(getMatrix(newParams.M1, newParams.N1))
      if ((globalThis.history) != undefined) {
        history.pushState(null, null, `?M1=${newParams.M1}&N1=${newParams.N1}&X1=${newParams.X1}`)
      }
    }
  }

  try {    
    if (!matrix.rows.allIds.length)  {
      return (
        <div className="container">
            <FormParamsMatrix addParamsHandle={addParamsHandle}/>
          </div>
      );
    }
    return (
      <div className="container">
        <Matrix matrix={matrix} />
        <AddRow addRowHandle={addRowHandle} />
      </div>
    );
  } catch (err) {
    console.log(err.message);
  }
};

const mapStateToProps = (state: IStateMatrix) => {
  return {
    matrix: state.matrix,
    params: state.params
  };
};

const mapDispatchToProps = {
  // addRow, addParams, addMatrix
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
