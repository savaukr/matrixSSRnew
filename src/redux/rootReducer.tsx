import {combineReducers} from 'redux'
import {matrixReducer} from './matrixReducer'
import {paramsReducer} from './paramsReducer'

export const rootReducer = combineReducers({
	matrix: matrixReducer,
	params: paramsReducer
})

export type RootState = ReturnType<typeof rootReducer>;