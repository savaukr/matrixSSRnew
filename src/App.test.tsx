import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from 'react-dom/test-utils';
import {act as actUtils} from 'react-dom/test-utils';
import { Provider } from "react-redux";
import { createStore } from 'redux'
import {rootReducer} from './redux/rootReducer'
import { act, create } from "react-test-renderer";
import {IStateMatrix} from './typesTS/typesTS'
import App from './App'

let container, store,
component, createComponent, renderComponent

//================================================================================
//test: are params that matrix renders, also form renders
describe("App component", () => {
    
    //------------------------------------------------------------------------
    test("it shows the form for adding params of matrix", () => {
        let preloadedState:IStateMatrix = {
            matrix: {matrix: []},
            params:  {M1:null, N1:null, X1:null}
        }
        store = createStore(rootReducer, preloadedState);
        
        act(() => {
            component = create(
                <Provider store={store}>
                    <App />
                </Provider>
            );
        })
        act(() => expect({M1:null, N1:null, X1:null}).toEqual(store.getState().params))
        expect(component.toJSON()).toMatchSnapshot()
      
        preloadedState = {
            matrix: {matrix:[[
                {amount: 100, bright: false, id: '0x0', part: false},
                {amount: 300, bright: false, id: '0x1', part: false},
                {amount: 440, bright: false, id: '0x2', part: false}
            ]]},
            params:  {M1:1, N1:3, X1:2}
        }
        store = createStore(rootReducer, preloadedState);
        act(() => {
            component.update(
                <Provider store={store}>
                    <App />
                </Provider>
            );
        })
        act(() => expect(preloadedState).toEqual(store.getState()))
        expect(component.toJSON()).toMatchSnapshot()
    });

    //--------------------------------------------------------------------
    test("send params", () => {
        let preloadedState:IStateMatrix = {
            matrix: {matrix: []},
            params:  {M1:null, N1:null, X1:null}
        }
        store = createStore(rootReducer, preloadedState);
        
        act(() => {
            component = create(
                <Provider store={store}>
                    <App />
                </Provider>
            );
        })
    
        const instance = component.root;
        const M1 = instance.findByProps({id: "M1"})
        const N1 = instance.findByProps({id: "N1"})
        const X1 = instance.findByProps({id: "X1"})
        
        expect(M1.props.value).toEqual('');
        const mEvent = { target: { value: '3' , name: 'M1'} };
        const nEvent ={ target: { value: '4' , name: 'N1'} };
        const xEvent = { target: { value: '2' , name: 'X1'} };
        act(() => {
            M1.props.onChange(mEvent)
        })
        act(() => { 
            N1.props.onChange(nEvent)
        })
        act(() => {
            X1.props.onChange(xEvent)
        })
        const form = instance.findByProps({id: "form"})
        const fEvent = {preventDefault: ()=>{}}
        act(() => {
            form.props.onSubmit(fEvent)
        })
        expect(store.getState().params).toEqual({ M1: 3, N1: 4, X1: 2 });
        

        // renderComponent = () => {
        //     ReactDOM.render(
        //         <Provider store={store}>
        //             <App />
        //         </Provider>, container);
        // }
        // actUtils(renderComponent);
        // const M1DOM = container.querySelector("#M1")
        // M1DOM.value = '3'
        // console.log("M1=", M1DOM.value)
        // //M1.value = '2'
        // M1DOM.value = '3'
        // //ReactTestUtils.Simulate.change(M1DOM, { target: { value: "2" } as unknown as EventTarget });
        // ReactTestUtils.Simulate.keyPress(M1DOM, {key: "2", keyCode: 50, which: 50});
        // console.log('store=', store.getState().params)
        // const form = container.querySelector('form')
        // const M1DOM = container.querySelector("#M1")
        
        // ReactTestUtils.Simulate.submit(form);


        // const button = container.querySelector("form button")
        // act(() => expect(store.getState().params).toEqual({ M1: null, N1: null, X1: null }))
        // ReactTestUtils.Simulate.click(button)
        // act(() => expect(store.getState().params).toEqual({ M1: 0, N1: 0, X1: 0 }))
        
    })
});

//=============================================================================================================
describe('test for user event', () => {
   
    const preloadedState = {
        matrix: {matrix:[[
            {amount: 100, bright: false, id: '0x0', part: false},
            {amount: 300, bright: false, id: '0x1', part: false},
            {amount: 440, bright: false, id: '0x2', part: false}
        ]]},
        params:  {M1:1, N1:2, X1:1}
    }
    
    container = document.createElement("div");
    document.body.appendChild(container);
  
    beforeEach(() => {  
        store = createStore(rootReducer, preloadedState);
        createComponent = () => {
            component = create(
                <Provider store={store}>
                    <App />
                </Provider>
            );
        }
        renderComponent = () => {
            ReactDOM.render(
                <Provider store={store}>
                    <App />
                </Provider>, container);
        }
    });

    // afterEach(() => {
    //    //document.body.removeChild(container);
    //    //container = null;
    // });
    
   //--------------------------------------------------------------------------
   //click on ceil
    test("it shows the matrix and it clicks on the ceil", () => {        
        act(createComponent)
        const instance = component.root;
        const ceil = instance.findByProps({'data-id':'0x1'});
        actUtils(renderComponent);
        const ceilDom = container.querySelector("[data-id='0x1']")
        ReactTestUtils.Simulate.click(ceilDom)
        const amount = store.getState().matrix.matrix[0][1].amount
        act(() => expect(+ceil.props.children[0]).toEqual(amount))
        expect(component.toJSON()).toMatchSnapshot()
    });
    //--------------------------------------------------------------------------------------
    //mouseover and mpuseout  on ceil
    test("it shows the matrix and it hover on the ceil", () => {        
        act(createComponent)
        actUtils(renderComponent);
        const ceil = container.querySelector("[data-id='0x1']")
        ReactTestUtils.Simulate.mouseOver(ceil)
        const brightOver = store.getState().matrix.matrix[0][0].bright
        act(() => expect(brightOver).toBe(true))
        ReactTestUtils.Simulate.mouseOut(ceil)
        const brightOut = store.getState().matrix.matrix[0][0].bright
        act(() => expect(brightOut).toBe(false))
        expect(component.toJSON()).toMatchSnapshot()
    });
    //--------------------------------------------------------------------------------------
    //hover on sum
    test("it shows the matrix and it hover on the sum", () => {        
        act(createComponent)
        actUtils(renderComponent);
        const ceilSum = container.querySelector("[data-ind='0']")
        ReactTestUtils.Simulate.mouseOver(ceilSum)
        const partOver = store.getState().matrix.matrix[0][0].part
        act(() => expect(partOver).toBe(true))
        ReactTestUtils.Simulate.mouseOut(ceilSum)
        const partOut = store.getState().matrix.matrix[0][0].part
        act(() => expect(partOut).toBe(false))
        expect(component.toJSON()).toMatchSnapshot()
    });
    //------------------------------------------------------------------------------------
    //click on delete row
    test("delete row", () => {        
        act(createComponent)
        actUtils(renderComponent);
        const lengthStart = store.getState().matrix.matrix.length
        ReactTestUtils.Simulate.click(container.querySelector(".sidebar-row button"))
        const lengthEnd = store.getState().matrix.matrix.length
        act(() => expect(lengthStart).toBe(lengthEnd+1))
        expect(component.toJSON()).toMatchSnapshot()
    });
    //------------------------------------------------------------------------------------
    //click on add row
    test("add row", () => {        
        act(createComponent)
        expect(component.toJSON()).toMatchSnapshot()
        actUtils(renderComponent);
        const lengthStart = store.getState().matrix.matrix.length
        ReactTestUtils.Simulate.click(container.querySelector(".addrRow-wrap button"))
        const lengthEnd = store.getState().matrix.matrix.length
        //console.log('store', store.getState().matrix.matrix)
        act(() => expect(lengthStart).toBe(lengthEnd-1))
       
    });

})