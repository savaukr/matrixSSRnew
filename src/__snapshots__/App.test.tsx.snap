// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it shows the form for adding params of matrix 1`] = `
<div
  className="container"
>
  <div
    className="form_params"
  >
    <div>
      
    </div>
    <form
      id="form"
      onSubmit={[Function]}
    >
      <div
        className="input-field"
      >
        <label
          htmlFor="M"
        >
          Стрічки:
        </label>
        <input
          id="M1"
          name="M1"
          onChange={[Function]}
          placeholder="Введіть кількість стрічок"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <label
          htmlFor="N"
        >
          Стовбці:
        </label>
        <input
          id="N1"
          name="N1"
          onChange={[Function]}
          placeholder="Введіть кількість стовпчиків"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <label
          htmlFor="X"
        >
          X1:
        </label>
        <input
          id="X1"
          name="X1"
          onChange={[Function]}
          placeholder="Введіть X"
          type="text"
          value=""
        />
      </div>
      <button
        className="sendParams"
        onClick={[Function]}
      >
        Відправити параметри матриці
      </button>
    </form>
  </div>
</div>
`;

exports[`App component it shows the form for adding params of matrix 2`] = `
<div
  className="container"
>
  <div
    className="matrix-wrap"
  >
    <div
      className="matrix-content"
    >
      <h4>
        Matrix 
        1
        x
        3
      </h4>
      <div
        className="matrix-header"
      >
        Сума по рядку
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          300
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={0}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          840
        </div>
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        />
        <div
          className="matrix-ceil footer  "
          data-id="footer0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          300
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={1}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          840
        </div>
      </div>
    </div>
  </div>
  <div
    className="addrRow-wrap"
  >
    <button
      onClick={[Function]}
    >
      Створити рядок
    </button>
  </div>
</div>
`;

exports[`test for user event add row 1`] = `
<div
  className="container"
>
  <div
    className="matrix-wrap"
  >
    <div
      className="matrix-content"
    >
      <h4>
        Matrix 
        1
        x
        3
      </h4>
      <div
        className="matrix-header"
      >
        Сума по рядку
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={0}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        />
        <div
          className="matrix-ceil footer  "
          data-id="footer0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={1}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
    </div>
  </div>
  <div
    className="addrRow-wrap"
  >
    <button
      onClick={[Function]}
    >
      Створити рядок
    </button>
  </div>
</div>
`;

exports[`test for user event delete row 1`] = `
<div
  className="container"
>
  <div
    className="form_params"
  >
    <div>
      
    </div>
    <form
      id="form"
      onSubmit={[Function]}
    >
      <div
        className="input-field"
      >
        <label
          htmlFor="M"
        >
          Стрічки:
        </label>
        <input
          id="M1"
          name="M1"
          onChange={[Function]}
          placeholder="Введіть кількість стрічок"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <label
          htmlFor="N"
        >
          Стовбці:
        </label>
        <input
          id="N1"
          name="N1"
          onChange={[Function]}
          placeholder="Введіть кількість стовпчиків"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field"
      >
        <label
          htmlFor="X"
        >
          X1:
        </label>
        <input
          id="X1"
          name="X1"
          onChange={[Function]}
          placeholder="Введіть X"
          type="text"
          value=""
        />
      </div>
      <button
        className="sendParams"
        onClick={[Function]}
      >
        Відправити параметри матриці
      </button>
    </form>
  </div>
</div>
`;

exports[`test for user event it shows the matrix and it clicks on the ceil 1`] = `
<div
  className="container"
>
  <div
    className="matrix-wrap"
  >
    <div
      className="matrix-content"
    >
      <h4>
        Matrix 
        1
        x
        3
      </h4>
      <div
        className="matrix-header"
      >
        Сума по рядку
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={0}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        />
        <div
          className="matrix-ceil footer  "
          data-id="footer0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={1}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
    </div>
  </div>
  <div
    className="addrRow-wrap"
  >
    <button
      onClick={[Function]}
    >
      Створити рядок
    </button>
  </div>
</div>
`;

exports[`test for user event it shows the matrix and it hover on the ceil 1`] = `
<div
  className="container"
>
  <div
    className="matrix-wrap"
  >
    <div
      className="matrix-content"
    >
      <h4>
        Matrix 
        1
        x
        3
      </h4>
      <div
        className="matrix-header"
      >
        Сума по рядку
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={0}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        />
        <div
          className="matrix-ceil footer  "
          data-id="footer0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={1}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
    </div>
  </div>
  <div
    className="addrRow-wrap"
  >
    <button
      onClick={[Function]}
    >
      Створити рядок
    </button>
  </div>
</div>
`;

exports[`test for user event it shows the matrix and it hover on the sum 1`] = `
<div
  className="container"
>
  <div
    className="matrix-wrap"
  >
    <div
      className="matrix-content"
    >
      <h4>
        Matrix 
        1
        x
        3
      </h4>
      <div
        className="matrix-header"
      >
        Сума по рядку
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        >
          <button
            onClick={[Function]}
          >
            ×
          </button>
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil   "
          data-id="0x2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={0}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
      <div
        className="matrix-row"
      >
        <div
          className="matrix-ceil sidebar-row"
        />
        <div
          className="matrix-ceil footer  "
          data-id="footer0"
          data-part="12%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          100
          <div
            style={
              Object {
                "height": "24%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer1"
          data-part="36%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          301
          <div
            style={
              Object {
                "height": "72%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil footer  "
          data-id="footer2"
          data-part="52%"
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          440
          <div
            style={
              Object {
                "height": "104%",
              }
            }
          />
        </div>
        <div
          className="matrix-ceil sum"
          data-id="sum"
          data-ind={1}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          841
        </div>
      </div>
    </div>
  </div>
  <div
    className="addrRow-wrap"
  >
    <button
      onClick={[Function]}
    >
      Створити рядок
    </button>
  </div>
</div>
`;
